mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,15 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.purgeOrder = async (req, res) => {
+  try {
+    const result = await Order.findOneAndRemove({ _id: req.params.orderId });
```

Comment:
In the `purgeOrder` function, the `if (result)` condition is incorrect. If the order is found and removed, `result` will be the removed document, which is truthy. Therefore, the code will always return a 404 status with the message "Order not located". It should be `if (!result)` instead.